akka-actor/src/main/scala/akka/actor/Props.scala

```diff
@@ -215,12 +213,12 @@ final case class Props(deploy: Deploy, clazz: Class[_], args: immutable.Seq[Any]
   /**
    * Returns a new Props with the specified dispatcher set.
    */
-  def withDispatcher(d: String): Props = copy(deploy = deploy.copy(dispatcher = d))
+  def withDispatcher(d: String): Props = if (d == dispatcher) this else copy(deploy = deploy.copy(dispatcher = d))
```

Comment:
I think the test must be on `deploy.dispatcher` to account for `NoDispatcherGiven` vs `DefaultDispatcherId`